readthedocs/search/parse_json.py

```diff
@@ -101,9 +104,71 @@ def parse_content(content):
 
     # removing the starting text of each
     content = content.split('\n')
-    if len(content) > 1:  # there were \n
+    if remove_first_line and len(content) > 1:
         content = content[1:]
 
     # converting newlines to ". "
     content = '. '.join([text.strip().rstrip('.') for text in content])
     return content
+
+
+def _get_text_for_domain_data(desc_contents):
+    """Returns the text from the PyQuery object ``desc_contents``."""
+    # remove the 'dl', 'dt' and 'dd' tags from it
+    # because all the 'dd' and 'dt' tags are inside 'dl'
+    # and all 'dl' tags are already captured.
+    desc_contents.remove('dl')
+    desc_contents.remove('dt')
+    desc_contents.remove('dd')
+
+    # remove multiple spaces, new line characters and '¶' symbol.
+    docstrings = parse_content(desc_contents.text())
+    return docstrings
+
+
+def generate_domains_data_from_pyquery(body, fjson_filename):
+    """
+    Given a pyquery object, generate sphinx domain objects' docstrings.
+
+    Returns a dict with the generated data.
+    The returned dict is in the following form::
+
+        {
+            "domain-id-1": {
+                "signature": "domain_signature_1(*args, **kwargs),
+                "docstrings": "docstrings for the domain-id-1",
+            },
+            "domain-id-2": {
+                "signature": "domain_signature_2(*args, **kwargs),
+                "docstrings": "docstrings for the domain-id-2",
+            }
+        }
+    """
+
+    domain_data = {}
+    try:
+        dl_tags = body('dl')
+        for dl_tag in dl_tags:
```

Comment:
I do wonder if we should put the `try` a bit lower here, so that we can continue parsing if we hit an issue. Here if we hit an exception in one tag, it will stop all others from processing.